control-plane/connect-inject/container_init.go

```diff
@@ -142,12 +149,42 @@ func (h *Handler) containerInit(namespace corev1.Namespace, pod corev1.Pod) (cor
 		TProxyExcludeUIDs:          splitCommaSeparatedItemsFromAnnotation(annotationTProxyExcludeUIDs, pod),
 		ConsulDNSClusterIP:         consulDNSClusterIP,
 		EnvoyUID:                   envoyUserAndGroupID,
+		MultiPort:                  multiPort,
+		EnvoyAddressPort:           20000 + mpi.serviceIndex,
```

Comment:
I think you are right, it might be redundant from what we're doing in endpoint controller for the proxy service registration, I'll try removing this and running acceptance tests!